diff -r d07fcec5578b emu/Linux/emu
--- a/emu/Linux/emu	Sat Mar 13 15:04:35 2010 +0000
+++ b/emu/Linux/emu	Sun Mar 21 19:56:02 2010 +0200
@@ -29,6 +29,7 @@
 	freetype
 	math
 	draw
+	re2wrap
 
 	memlayer
 	memdraw
@@ -50,6 +51,7 @@
 	keyring
 	loader
 	freetype
+	re2
 
 port
 	alloc
diff -r d07fcec5578b emu/Linux/emu-g
--- a/emu/Linux/emu-g	Sat Mar 13 15:04:35 2010 +0000
+++ b/emu/Linux/emu-g	Sun Mar 21 19:56:02 2010 +0200
@@ -24,6 +24,7 @@
 lib
 	interp
 	math
+	re2wrap
 	keyring
 	sec
 	mp
@@ -38,6 +39,7 @@
 	srv	srv
 	keyring
 	loader
+	re2
 
 port
 	alloc
diff -r d07fcec5578b emu/Linux/mkfile
--- a/emu/Linux/mkfile	Sat Mar 13 15:04:35 2010 +0000
+++ b/emu/Linux/mkfile	Sun Mar 21 19:56:02 2010 +0200
@@ -33,7 +33,7 @@
 HFILES=\
 
 CFLAGS='-DROOT="'$ROOT'"' -DEMU -I. -I../port -I$ROOT/$SYSTARG/$OBJTYPE/include -I$ROOT/include -I$ROOT/libinterp $CTHREADFLAGS $CFLAGS $EMUOPTIONS
-SYSLIBS= $X11LIBS -lm
+SYSLIBS= $X11LIBS -lm -lre2 -lstdc++ -lpthread
 KERNDATE=`{$NDATE}
 
 default:V:	$O.$CONF
diff -r d07fcec5578b libinterp/mkfile
--- a/libinterp/mkfile	Sat Mar 13 15:04:35 2010 +0000
+++ b/libinterp/mkfile	Sun Mar 21 19:56:02 2010 +0200
@@ -31,6 +31,7 @@
 	das-$OBJTYPE.$O\
 	keyring.$O\
 	string.$O\
+	re2.$O\
 
 HFILES=\
 	$ROOT/include/interp.h\
@@ -48,6 +49,7 @@
 	../module/keyring.m\
 	../module/loader.m\
 	../module/freetype.m\
+	../module/re2.m\
 
 <$ROOT/mkfiles/mksyslib-$SHELLTYPE
 
@@ -78,6 +80,9 @@
 freetypemod.h:D: $MODULES
 	rm -f $target && limbo -t Freetype -I../module ../module/runt.m > $target
 
+re2mod.h:D: $MODULES
+	rm -f $target && limbo -t Re2 -I../module ../module/runt.m > $target
+
 benchmod.h:D: ../module/bench.m
 	rm -f $target && limbo -t Bench -I../module ../module/bench.m > $target
 
@@ -100,6 +105,7 @@
 loader.$O:	loadermod.h
 freetype.$O:	freetypemod.h $ROOT/include/freetype.h
 math.$O:	$ROOT/include/mathi.h
+re2.$O:		re2mod.h
 
 das-spim.c:N:	das-mips.c
 das-68000.c:N:	das-68020.c
diff -r d07fcec5578b module/runt.m
--- a/module/runt.m	Sat Mar 13 15:04:35 2010 +0000
+++ b/module/runt.m	Sun Mar 21 19:56:02 2010 +0200
@@ -9,3 +9,4 @@
 include "loader.m";
 # include "readimage.m";
 include "freetype.m";
+include "re2.m";
