diff -r d07fcec5578b emu/Linux/emu
--- a/emu/Linux/emu	Sat Mar 13 15:04:35 2010 +0000
+++ b/emu/Linux/emu	Sun Mar 21 19:56:02 2010 +0200
@@ -29,6 +29,7 @@
 	freetype
 	math
 	draw
+	re2wrap
 
 	memlayer
 	memdraw
@@ -52,6 +53,7 @@
 	ipints
 	loader
 	freetype
+	re2
 
 port
 	alloc
diff -r d07fcec5578b emu/Linux/emu-g
--- a/emu/Linux/emu-g	Sat Mar 13 15:04:35 2010 +0000
+++ b/emu/Linux/emu-g	Sun Mar 21 19:56:02 2010 +0200
@@ -24,6 +24,7 @@
 lib
 	interp
 	math
+	re2wrap
 	keyring
 	sec
 	mp
@@ -40,6 +41,7 @@
 	crypt
 	ipints
 	loader
+	re2
 
 port
 	alloc
diff -r d07fcec5578b emu/Linux/mkfile
--- a/emu/Linux/mkfile	Sat Mar 13 15:04:35 2010 +0000
+++ b/emu/Linux/mkfile	Sun Mar 21 19:56:02 2010 +0200
@@ -33,7 +33,7 @@
 HFILES=\
 
 CFLAGS='-DROOT="'$ROOT'"' -DEMU -I. -I../port -I$ROOT/$SYSTARG/$OBJTYPE/include -I$ROOT/include -I$ROOT/libinterp $CTHREADFLAGS $CFLAGS $EMUOPTIONS
-SYSLIBS= $X11LIBS -lm -lrt -lpthread
+SYSLIBS= $X11LIBS -lm -lrt -lpthread -lre2 -lstdc++
 KERNDATE=`{$NDATE}
 
 default:V:	$O.$CONF
diff -r d07fcec5578b libinterp/mkfile
--- a/libinterp/mkfile	Sat Mar 13 15:04:35 2010 +0000
+++ b/libinterp/mkfile	Sun Mar 21 19:56:02 2010 +0200
@@ -32,6 +32,7 @@
 	das-$OBJTYPE.$O\
 	keyring.$O\
 	string.$O\
+	re2.$O\
 
 HFILES=\
 	$ROOT/include/interp.h\
@@ -52,6 +53,7 @@
 	../module/ipints.m\
 	../module/crypt.m\
 	keyringif.m\
+	../module/re2.m\
 
 <$ROOT/mkfiles/mksyslib-$SHELLTYPE
 
@@ -85,6 +87,9 @@
 ipintsmod.h:D:	$MODULES
 	rm -f $target && limbo -t IPints -I../module ../module/ipints.m > $target
 
+re2mod.h:D: $MODULES
+	rm -f $target && limbo -t Re2 -I../module ../module/runt.m > $target
+
 benchmod.h:D: ../module/bench.m
 	rm -f $target && limbo -t Bench -I../module ../module/bench.m > $target
 
@@ -115,6 +120,7 @@
 loader.$O:	loadermod.h
 freetype.$O:	freetypemod.h $ROOT/include/freetype.h
 math.$O:	$ROOT/include/mathi.h
+re2.$O:		re2mod.h
 
 das-spim.c:N:	das-mips.c
 das-68000.c:N:	das-68020.c
diff -r d07fcec5578b module/runt.m
--- a/module/runt.m	Sat Mar 13 15:04:35 2010 +0000
+++ b/module/runt.m	Sun Mar 21 19:56:02 2010 +0200
@@ -9,3 +9,4 @@
 include "crypt.m";
 include "loader.m";
 include "freetype.m";
+include "re2.m";
